### ISSUE TITLE
[Feature Request]: Allow for passing custom label for page select in Pagination

### ISSUE THREAD

**Issue by itspatys** [carbon-design-system/carbon#18516]
### The problem

The `Pagination` component doesn't allow for passing custom label to page selector. The value is hard-coded (https://github.com/carbon-design-system/carbon/blob/fa1abc1b47b4daa5c52c9e9d8b03fdd65780e0d8/packages/react/src/components/Pagination/Pagination.tsx#L391).

### The solution

A prop to `Pagination` component could be introduced that would allow for providing custom label. It could accept a function with `totalPages` as argument.

### Business priority

Medium Priority = upcoming release but is not pressing

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)

### Tasks

- [ ] React Implementation
- [ ] Web Components Implementation

**Comment by tay1orjones** [comment]
It's strange and unexpected to find this isn't translatable, but looking back in the history it appears it hasn't been for [a long time](https://github.com/carbon-design-system/carbon/blame/8379ffcce6e07fb95cf55cbf1de5b91b893d4a62/packages/react/src/components/Pagination/Pagination.js#L316). At one point it was though - https://github.com/carbon-design-system/carbon/commit/dc35195b3f613d52caf46daf3c26a272629c9573

Agree this should be a function as the other translatable strings are in there, `pageRangeText`, `pageText`, etc.

**Comment by tay1orjones** [comment]
This technically only impacts screenreader users

**Comment by kadiryazgan** [comment]
Sent PR #19839 for this isse
